### PR TITLE
refactor, cleanup, polish

### PR DESCRIPTION
**Pull request by pvuorela**
Bunch of changes to make connectionagent a bit more readable. Functionality shouldn't be much changed except by avoiding memory leak and race condition on instance(). Tests didn't seem to pass too well even before, and are quite dependent on execution environment, so mainly kept them compiling.

### CODE REVIEW

**Comment by pvuorela** [comment]
@monich or @lpotter ?

File: connd/wakeupwatcher.cpp [hunk 1]
```diff
@@ -19,6 +19,18 @@
...
+#define MCE_SIGNAL_PATH                 "/com/nokia/mce/signal"
+#define MCE_SIGNAL_INTERFACE            "com.nokia.mce.signal"
+#define MCE_PSM_STATE_IND               "psm_state_ind"
+#define MCE_DISPLAY_IND                 "display_status_ind"
+
+#define MCE_REQUEST_PATH                "/com/nokia/mce/request"
+#define MCE_REQUEST_INTERFACE           "com.nokia.mce.request"
+#define MCE_DISPLAY_STATUS_GET          "get_display_status"
+#define MCE_PSM_STATE_GET               "get_psm_state"
+
```
Comment (monich) [inline connd/wakeupwatcher.cpp]: I would suggest to pull those from `<mce/dbus-names.h>` and `<mce/mode-names.h>` headers provided by `mce-headers`package.

**Comment by monich** [comment]
Haven't tested it but generally looks OK to me.

File: connd/wakeupwatcher.cpp (hunk 1 above)
Comment (pvuorela) [inline connd/wakeupwatcher.cpp]: That would add build dependency here and I'm not sure if it's justified for this small amount of strings.

Comment (monich) [inline connd/wakeupwatcher.cpp]: I'm not sure either, it's up to you.

Comment (pvuorela) [inline connd/wakeupwatcher.cpp]: I'll skip that now. Benefit is not clear and here I merely moved lines from file to another.

Comment (lpotter) [inline connd/wakeupwatcher.cpp]: I originally added these here, so I didn't need mce headers to be installed just to build this, as it does not link to mce at runtime.

**Pull request merged**